docs/src/reference/gremlin-variants.asciidoc

```diff
@@ -1897,3 +1897,299 @@ pip install gremlinpython
+|NewConnectionThreshold | Minimum amount of concurrent active traversals on a connection to trigger creation of a new connection. |4
+|MaximumConcurrentConnections | Maximum number of concurrent connections. |number of runtime processors
+|EnableCompression |Flag to enable compression. |false
+|ReadBufferSize |Specify I/O buffer sizes in bytes. If a buffer size is zero, then a useful default size is used |0
+|WriteBufferSize |Specify I/O buffer sizes in bytes. If a buffer size is zero, then a useful default size is used |0
+|Session |Session ID. |""
+|=========================================================
+
+[[gremlin-go-strategies]]
+=== Traversal Strategies
+
+In order to add and remove <<traversalstrategy,traversal strategies>> from a traversal source, Gremlin-Go has a
+`TraversalStrategy` interface along with a collection of functions that mirror the standard Gremlin-Java strategies.
+
+[source,go]
+----
+promise := g.WithStrategies(ReadOnlyStrategy()).AddV("person").Property("name", "foo").Iterate()
+----
+
+NOTE: Many of the `TraversalStrategy` classes in Gremlin-Go are proxies to the respective strategy on
+Apache TinkerPop's JVM-based Gremlin traversal machine. As such, their `apply(Traversal)` method does nothing. However,
+the strategy is encoded in the Gremlin-Go bytecode and transmitted to the Gremlin traversal machine for
+re-construction machine-side.
+
+[[gremlin-go-transactions]]
+=== Transactions
+
+To get a full understanding of this section, it would be good to start by reading the <<transactions,Transactions>>
+section of this documentation, which discusses transactions in the general context of TinkerPop itself. This section
+builds on that content by demonstrating the transactional syntax for Go.
+
+[source,go]
+----
+remote, err := NewDriverRemoteConnection("ws://localhost:8182/gremlin")
+g := Traversal_().WithRemote(remote)
+
+// Create a Transaction.
+tx := g.Tx()
+
+// Spawn a new GraphTraversalSource, binding all traversals established from it to tx.
+gtx, _ := tx.Begin()
+
+// Execute a traversal within the transaction.
+promise := g.AddV("person").Property("name", "Lyndon").Iterate()
+err := <-promise
+
+if err != nil {
+  // Rollback the transaction if an error occurs.
+  tx.rollback()
+} else {
+  // Commit the transaction. The transaction can no longer be used and cannot be re-used.
+  // A new transaction can be spawned through g.Tx().
+  tx.Commit()
+}
+----
+
+[[gremlin-go-lambda]]
+=== The Lambda Solution
+
+Supporting link:https://en.wikipedia.org/wiki/Anonymous_function[anonymous functions] across languages is difficult as
+most languages do not support lambda introspection and thus, code analysis. In Gremlin-Go, a Gremlin lambda should
+be represented as a zero-arg callable that returns a string representation of the lambda expected for use in the
+traversal. The lambda should be written as a `Gremlin-Groovy`string. When the lambda is represented in `Bytecode` its
```

Comment:
Nit:  `Gremlin-Groovy`string appears to be missing a space.

Reply:
fixed both https://github.com/lyndonbauto/tinkerpop/pull/94/commits/453b5b6f32cf87d27d618a1d968fe78ff26a4d6b